### PR TITLE
Threaded comments - Added ability to reply to comments and view comments as a threaded list.

### PR DESCRIPTION
**Pull request by athifw**

### CODE REVIEW

File: app/views/articles/show.html.erb [hunk 1]
```diff
@@ -2,7 +2,24 @@
...
 <%= link_to 'Back', articles_path %>
 <%= link_to 'New Comment', new_article_comment_path(@article) %>
-<%= link_to 'Show Comments', article_comments_path(@article) %>
+<br>
 
+<% def recursion(old_comment) %>
+  <% old_comment.comments.each do |comment| %>
+  <br>
+    <li><%= render 'comments/view', comment: comment %></li>
+    <% if !comment.comments.empty? %>
```
Comment (dce) [inline app/views/articles/show.html.erb]: `if comment.comments.any?`

File: app/views/articles/show.html.erb [hunk 2]
```diff
@@ -2,7 +2,24 @@
 <br>
 <%= link_to 'Back', articles_path %>
 <%= link_to 'New Comment', new_article_comment_path(@article) %>
-<%= link_to 'Show Comments', article_comments_path(@article) %>
+<br>
 
+<% def recursion(old_comment) %>
```
Comment (dce) [inline app/views/articles/show.html.erb]: Whoa. You're defining a method inside a view? What does this do?

File: app/views/comments/_view.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<%= comment.user %>
+<br>
+<%= comment %>
+<br>
+Posted <%= time_ago_in_words(comment.created_at) %> ago
+<br>
+<%= link_to 'Reply', new_comment_comment_path(comment) %>
+<br>
```
Comment (dce) [inline app/views/comments/_view.html.erb]: I'd probably call this `_comment.html.erb` -- calling a view "view" is weird.

File: db/migrate/20150706183824_change_comments_to_polymorphic.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class ChangeCommentsToPolymorphic < ActiveRecord::Migration
+  def up
+    rename_column :comments, :article_id, :commentable_id
+    add_column :comments, :commentable_type, :string
+
+    execute "UPDATE comments SET commentable_type = 'Article'"
```
Comment (dce) [inline db/migrate/20150706183824_change_comments_to_polymorphic.rb:6]: Mark this column non-null after you set this default?

File: spec/models/comment_spec.rb [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
-      expect(subject.to_s).to eq("Test comment")
+      new_comment.text = "Test comment"
+      expect(new_comment.to_s).to eq("Test comment")
+    end
+  end
+
+  describe "article" do
+    it "returns article commented on" do
+      expect(new_comment.article.to_s).to eq('Comment tester')
     end
```
Comment (dce) [inline spec/models/comment_spec.rb:24]: You don't really test the recursion here, just the base case.

File: app/views/articles/show.html.erb (hunk 2 above)
Comment (dce) [inline app/views/articles/show.html.erb]: Ok, I see. I'd just have the comment partial recursively render itself.

**Comment by yaychris** [comment]
Looks good :dragon:

**Pull request closed without merging**